### PR TITLE
Ncnews 3

### PR DESCRIPTION
**Pull request by javidraeisy**

### CODE REVIEW

File: __tests__/apiTopics.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+beforeEach(() => {
+  return seed(data);
+});
+describe("Get / api/topics", () => {
+  it("returns with array of objects in topics", () => {
+    return request(app)
+      .get("/api/topics")
+      .then((response) => {
+        expect(response.status).toBe(200);
+        expect(response.body).toBeInstanceOf(Array);
```
Comment (itskatherine) [inline __tests__/apiTopics.test.js:20]: Whenever we use a forEach loop, it's important we also check the length of the array - as this test could otherwise pass if it were being passed an empty array (as the forEach loop wouldn't happen), please add in a length check for the array

File: __tests__/apiTopics.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  db.end();
+});
+
+beforeEach(() => {
+  return seed(data);
+});
+describe("Get / api/topics", () => {
+  it("returns with array of objects in topics", () => {
+    return request(app)
+      .get("/api/topics")
```
Comment (itskatherine) [inline __tests__/apiTopics.test.js:17]: you can chain on a  `.expect(200)` here instead of needing line 19 i.e.

`return request(app).get("/api/topics").expect(200).then((response)....etc....`

File: controllers/getAllTopicsController.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const getAllTopics = require("../models/getAllTopicsModel");
+
+async function getAllTopicsController(req, res, next) {
+  try {
+    const topics = await getAllTopics();
+    return res.status(200).send(topics.rows);
```
Comment (itskatherine) [inline controllers/getAllTopicsController.js:6]: The best place to extract the data off the rows key is in the model, as it's data manipulation, which is the model's thing

File: __tests__/apiTopics.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      .then((response) => {
+        expect(response.status).toBe(200);
+        expect(response.body).toBeInstanceOf(Array);
+        response.body.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+        });
+      });
+  });
+});
```
Comment (itskatherine) [inline __tests__/apiTopics.test.js:27]: Great test, can you add a sad path test for if the user spells the path wrong, for example `/api/toopics` 
This should give a 404 and an error message

File: index.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const express = require("express");
+
+const app = express();
+
+const topicsRouter = require("./routes/topics");
+
+app.use(express.json());
```
Comment (itskatherine) [inline index.js:7]: We don't need this line yet as we only need this line to access the body for a get request - it'll be needed once we get to our post request later. As its best practise to only include code that meets the need of the ticket you're working on, this should be removed in this PR

**Review by itskatherine — COMMENTED** [review]
Hi Javid, nice work - please have a look over my comments and once you've addressed them feel free to merge 👍
